src/icrar/leap-accelerate/math/casacore_helper.h

```diff
@@ -91,6 +91,7 @@ namespace icrar
         for(T t : value)
         {
             *result_it = lambda(t);
+            result_it++;
         }
```

Comment:
This looks like `std::transform`, where you wouldn't need to keep track of `result_it`.

In fact, the first two `MapCollection` overloads look very similar to `vector_map`, and could probably be implemented similarly.